exchanges/poloniex/poloniex_test.go

```diff
@@ -141,901 +84,2500 @@ func TestGetFee(t *testing.T) {
+		Amount:       10000000,
+		ClientID:     "hi",
+		AssetType:    asset.Futures,
+	})
+	require.NoError(t, err)
+	assert.NotNil(t, result)
+
+	result, err = e.SubmitOrder(generateContext(), &order.Submit{
+		Exchange:     e.Name,
+		Pair:         futuresTradablePair,
+		Side:         order.Buy,
+		Type:         order.TrailingStop,
+		TriggerPrice: 11,
+		Price:        10,
+		Amount:       10000000,
+		AssetType:    asset.Futures,
+	})
+	require.NoError(t, err)
+	assert.NotNil(t, result)
 }
 
-func TestGetOrderTrades(t *testing.T) {
+func TestWebsocketSubmitOrder(t *testing.T) {
 	t.Parallel()
+	arg := &order.Submit{AssetType: asset.Futures, TimeInForce: order.GoodTillCrossing}
+	_, err := e.WebsocketSubmitOrder(t.Context(), arg)
+	require.ErrorIs(t, err, currency.ErrCurrencyPairEmpty)
 
-	tests := []struct {
-		name           string
-		mock           bool
-		orderID        string
-		errExpected    bool
-		errMsgExpected string
-	}{
-		{
-			name:           "correct order ID",
-			mock:           true,
-			orderID:        "96238912841",
-			errExpected:    false,
-			errMsgExpected: "",
-		},
-		{
-			name:           "wrong order ID",
-			mock:           true,
-			orderID:        "96238912842",
-			errExpected:    true,
-			errMsgExpected: "Order not found",
-		},
-	}
+	arg.Pair = futuresTradablePair
+	_, err = e.WebsocketSubmitOrder(t.Context(), arg)
+	require.ErrorIs(t, err, order.ErrInvalidTimeInForce)
 
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			t.Parallel()
-			if tt.mock != mockTests {
-				t.Skip("mock mismatch, skipping")
-			}
-
-			_, err := e.GetAuthenticatedOrderTrades(t.Context(), tt.orderID)
-			switch {
-			case sharedtestvalues.AreAPICredentialsSet(e) && err != nil:
-				t.Errorf("Could not get order trades: %s", err)
-			case !sharedtestvalues.AreAPICredentialsSet(e) && err == nil && !mockTests:
-				t.Error("Expecting an error when no keys are set")
-			case mockTests && err != nil:
-				assert.ErrorContains(t, err, tt.errMsgExpected)
-			}
-		})
-	}
-}
+	arg.TimeInForce = order.GoodTillCancel
+	arg.AssetType = asset.Options
+	_, err = e.WebsocketSubmitOrder(t.Context(), arg)
+	require.ErrorIs(t, err, asset.ErrNotSupported)
 
-// Any tests below this line have the ability to impact your orders on the exchange. Enable canManipulateRealOrders to run them
-// ----------------------------------------------------------------------------------------------------------------------------
+	arg.AssetType = asset.Spot
+	arg.Type = order.Liquidation
+	arg.Pair = spotTradablePair
+	_, err = e.WebsocketSubmitOrder(t.Context(), arg)
+	require.ErrorIs(t, err, order.ErrUnsupportedOrderType)
 
-func TestSubmitOrder(t *testing.T) {
-	t.Parallel()
-	if !mockTests {
-		sharedtestvalues.SkipTestIfCannotManipulateOrders(t, e, canManipulateRealOrders)
+	e := new(Exchange) //nolint:govet // Intentional shadow
+	require.NoError(t, testexch.Setup(e), "Test instance Setup must not error")
+
+	if mockTests {
+		t.SkipNow()
```

Comment:
There are a lot of t.SkipNow() without messages, these require messages to say why its being introduced, so that in future it might be removed with the correct mock for example. 🌴